pkgs/shells/rc/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, fetchurl, autoreconfHook
+, ncurses #acinclude.m4 wants headers for tgetent().
+, historySupport ? false
+, editSupport ? "readline", readline ? null, libedit ? null
+}:
+
+assert editSupport == "readline" -> readline != null;
```

Comment:
@RamKromberg those all sound like internal details to me. Your last example illustrates quite well where reifying every option leads to and why we should avoid doing so.

Please consider that expression parameters have very poor discoverability, they shouldn't be treated like gentoo USE flags.